### ISSUE TITLE
[saas-18] When created a new database, if not clean the cookies an error 500 appears.

### ISSUE THREAD

**Issue by nhomar** [odoo/odoo#19659]
Impacted versions: **saas-18**

Steps to reproduce:

- Install a new database in a pretty new instlallation.
- Delete you Database if exists.
- Init the server with parameter -d dbname -i module.
- Wait until all installed. (no errors should be found).
- Go to the same browser (Chrome on my case) "A traceback appears"
- Go to the same browser on incognito mode, "Login screen is shown".
- Clear the coockies in the old browser "It works"

Current behavior:

Traceback appears when cookies not cleaned in a new database with the same old name in the same server.

```python
7 13:49:41] "GET /web HTTP/1.1" 500 -
2017-09-22 13:49:41,998 23343 ERROR saas18 werkzeug: Error on request:
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/werkzeug/serving.py", line 193, in run_wsgi
    execute(self.server.app)
  File "/usr/local/lib/python3.5/dist-packages/werkzeug/serving.py", line 181, in execute
    application_iter = app(environ, start_response)
  File "/home/nhomar/odoo/odoo/service/server.py", line 251, in app
    return self.app(e, s)
  File "/home/nhomar/odoo/odoo/service/wsgi_server.py", line 155, in application
    return application_unproxied(environ, start_response)
  File "/home/nhomar/odoo/odoo/service/wsgi_server.py", line 143, in application_unproxied
    result = handler(environ, start_response)
  File "/home/nhomar/odoo/odoo/http.py", line 1300, in __call__
    return self.dispatch(environ, start_response)
  File "/home/nhomar/odoo/odoo/http.py", line 1274, in __call__
    return self.app(environ, start_wrapped)
  File "/usr/local/lib/python3.5/dist-packages/werkzeug/wsgi.py", line 599, in __call__
    return self.app(environ, start_response)
  File "/home/nhomar/odoo/odoo/http.py", line 1474, in dispatch
    result = ir_http._dispatch()
  File "/home/nhomar/odoo/addons/web_editor/models/ir_http.py", line 21, in _dispatch
    return super(IrHttp, cls)._dispatch()
  File "/home/nhomar/odoo/odoo/addons/base/ir/ir_http.py", line 212, in _dispatch
    return cls._handle_exception(e)
  File "/home/nhomar/odoo/odoo/addons/base/ir/ir_http.py", line 182, in _handle_exception
    return request._handle_exception(exception)
  File "/home/nhomar/odoo/odoo/http.py", line 763, in _handle_exception
    return super(HttpRequest, self)._handle_exception(exception)
  File "/home/nhomar/odoo/odoo/http.py", line 309, in _handle_exception
    raise pycompat.reraise(type(exception), exception, sys.exc_info()[2])
  File "/home/nhomar/odoo/odoo/tools/pycompat.py", line 86, in reraise
    raise value
  File "/home/nhomar/odoo/odoo/addons/base/ir/ir_http.py", line 208, in _dispatch
    result = request.dispatch()
  File "/home/nhomar/odoo/odoo/http.py", line 822, in dispatch
    r = self._call_function(**self.params)
  File "/home/nhomar/odoo/odoo/http.py", line 341, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "/home/nhomar/odoo/odoo/service/model.py", line 97, in wrapper
    return f(dbname, *args, **kwargs)
  File "/home/nhomar/odoo/odoo/http.py", line 337, in checked_call
    result.flatten()
  File "/home/nhomar/odoo/odoo/http.py", line 1251, in flatten
    self.response.append(self.render())
  File "/home/nhomar/odoo/odoo/http.py", line 1244, in render
    return env["ir.ui.view"].render_template(self.template, self.qcontext)
  File "/home/nhomar/odoo/odoo/addons/base/ir/ir_ui_view.py", line 1047, in render_template
    return self.browse(self.get_view_id(template)).render(values, engine)
  File "/home/nhomar/odoo/addons/web_editor/models/ir_ui_view.py", line 27, in render
    return super(IrUiView, self).render(values=values, engine=engine)
  File "/home/nhomar/odoo/odoo/addons/base/ir/ir_ui_view.py", line 1056, in render
    return self.env[engine].render(self.id, qcontext)
  File "/home/nhomar/odoo/odoo/addons/base/ir/ir_qweb/ir_qweb.py", line 57, in render
    return super(IrQWeb, self).render(id_or_xml_id, values=values, **context)
  File "/home/nhomar/odoo/odoo/addons/base/ir/ir_qweb/qweb.py", line 271, in render
    self.compile(template, options)(self, body.append, values or {})
  File "/home/nhomar/odoo/odoo/addons/base/ir/ir_qweb/qweb.py", line 348, in _compiled_fn
    raise QWebException("Error to render compiling AST", e, path, node and etree.tostring(node[0], encoding='unicode'), name)
odoo.addons.base.ir.ir_qweb.qweb.QWebException: sequence item 0: expected str instance, bytes found
Traceback (most recent call last):
  File "/home/nhomar/odoo/odoo/tools/cache.py", line 84, in lookup
    r = d[key]
  File "/home/nhomar/odoo/odoo/tools/func.py", line 68, in wrapper
    return func(self, *args, **kwargs)
  File "/home/nhomar/odoo/odoo/tools/lru.py", line 44, in __getitem__
    a = self.d[obj].me
KeyError: ('ir.qweb', <function IrQWeb._get_asset at 0x7f51d6f1e488>, 'web.assets_backend', 'en_US', True, False, False, False, (None,))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/nhomar/odoo/odoo/addons/base/ir/ir_qweb/qweb.py", line 341, in _compiled_fn
    return compiled(self, append, new, options, log)
  File "<template>", line 1, in template_166_71
  File "<template>", line 2, in body_call_content_70
  File "<template>", line 3, in set_68
  File "<decorator-gen-25>", line 2, in _get_asset
  File "/home/nhomar/odoo/odoo/tools/cache.py", line 89, in lookup
    value = d[key] = self.method(*args, **kwargs)
  File "/home/nhomar/odoo/odoo/addons/base/ir/ir_qweb/ir_qweb.py", line 197, in _get_asset
    return asset.to_html(css=css, js=js, debug=debug, async=async, url_for=(values or {}).get('url_for', lambda url: url))
  File "/home/nhomar/odoo/odoo/addons/base/ir/ir_qweb/assetsbundle.py", line 122, in to_html
    css_attachments = self.css() or []
  File "/home/nhomar/odoo/odoo/addons/base/ir/ir_qweb/assetsbundle.py", line 240, in css
    css = self.preprocess_css()
  File "/home/nhomar/odoo/odoo/addons/base/ir/ir_qweb/assetsbundle.py", line 341, in preprocess_css
    compiled = self.compile_css(cmd, source)
  File "/home/nhomar/odoo/odoo/addons/base/ir/ir_qweb/assetsbundle.py", line 403, in compile_css
    cmd_output = ''.join(result)
TypeError: sequence item 0: expected str instance, bytes found

Error to render compiling AST
TypeError: sequence item 0: expected str instance, bytes found
Template: 166
Path: /templates/t/t/t[1]/t[2]
Node: <t t-call-assets="web.assets_backend" t-js="false"/>
```

Expected behavior:

- No error, simply clean the cookies and show the login screen again.

Video/Screenshot link (optional):

**Comment by nhomar** [comment]
I did not log into the database but when done, the error appears again even if the cookies was cleaned.

Then the -i module is unusable offically. :-s

**Comment by nhomar** [comment]
To compliment the command I am running is:

```
python3 ./odoo-bin --addons-path=../enterprise,addons -i base -d saas18
```

**Comment by nhomar** [comment]
i could not reproducce the error anymore, I deleted the /home/nhomar/.local/share/Odoo content, may be a cache was messing my environment.

**Comment by odony** [comment]
Actually I managed to reproduce a similar case, thanks for reporting it. You will get this unusual trace when the LESS compilation fails, and the system is trying to parse the error in P3. This is a P3 conversion bug, which should be fixed after f565a18cad26dd5cede9eb6731908b1ae51ff1c7.

The root cause for the failed parsing is probably some backwards-incompatible change we did in saas-17/18, which breaks the LESS compilation after a git pull. Re-creating the database usually fixes it.
